adaptive_ai/adaptive_ai/core/config.py

```
@@ -1,90 +1,264 @@
+    def _load_yaml_config(self) -> None:
+        """Load configuration from YAML file."""
+        config_path = self.get_config_file_path()
+        if config_path and config_path.exists():
+            try:
+                yaml_config = _load_yaml_config(str(config_path))
+                self._merge_yaml_config(yaml_config)
+            except Exception as e:
+                # Log warning but don't fail - fall back to defaults
+                print(f"Warning: Could not load config file {config_path}: {e}")
+
+    def _merge_yaml_config(self, yaml_config: Dict[str, Any]) -> None:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)